tests/light/functional_tests/filterx/test_filterx_update_metric.py

```diff
@@ -0,0 +1,189 @@
+
+
+def create_config(config, port_allocator, filterx, stats_level=0):
+    network_source = config.create_network_source(port=port_allocator(), flags="no-parse")
+    file_destination = config.create_file_destination(file_name="output.log")
+    raw_config = f"""
+@version: {config.get_version()}
+
+options {{ stats(level({stats_level})); }};
+
+source s_network {{
+    {render_statement(network_source)};
+}};
+
+destination d_file {{
+    {render_statement(file_destination)};
+}};
+
+log {{
+    source(s_network);
+    filterx {{
+        {filterx}
+    }};
+    destination(d_file);
+}};
+"""
+    config.set_raw_config(raw_config)
+    return network_source, file_destination
+
+
+def test_filterx_update_metric_labels(config, port_allocator, syslog_ng):
+    network_source, file_destination = create_config(
+        config,
+        port_allocator,
+        r"""
+            update_metric("literal", labels={"msg": $MSG, "foo": "foovalue"});
+
+            labels = json();
+            labels.msg = $MSG;
+            labels.foo = "foovalue";
+            update_metric("non_literal", labels=labels);
+        """,
+    )
+
+    syslog_ng.start(config)
+    network_source.write_log("msg1\nmsg2\nmsg1\nmsg3\n")
+    file_destination.read_logs(4)
+
+    # literal
+
+    samples = config.get_prometheus_samples([MetricFilter("syslogng_literal", {})])
+    assert len(samples) == 3
+
+    samples = config.get_prometheus_samples([MetricFilter("syslogng_literal", {"msg": "msg1", "foo": "foovalue"})])
+    assert len(samples) == 1
+    assert int(samples[0].value) == 2
+
+    samples = config.get_prometheus_samples([MetricFilter("syslogng_literal", {"msg": "msg2", "foo": "foovalue"})])
+    assert len(samples) == 1
+    assert int(samples[0].value) == 1
+
+    samples = config.get_prometheus_samples([MetricFilter("syslogng_literal", {"msg": "msg3", "foo": "foovalue"})])
+    assert len(samples) == 1
+    assert int(samples[0].value) == 1
+
+    # non literal
+
+    samples = config.get_prometheus_samples([MetricFilter("syslogng_non_literal", {})])
+    assert len(samples) == 3
+
+    samples = config.get_prometheus_samples([MetricFilter("syslogng_non_literal", {"msg": "msg1", "foo": "foovalue"})])
+    assert len(samples) == 1
+    assert int(samples[0].value) == 2
+
+    samples = config.get_prometheus_samples([MetricFilter("syslogng_non_literal", {"msg": "msg2", "foo": "foovalue"})])
+    assert len(samples) == 1
+    assert int(samples[0].value) == 1
+
+    samples = config.get_prometheus_samples([MetricFilter("syslogng_non_literal", {"msg": "msg3", "foo": "foovalue"})])
+    assert len(samples) == 1
+    assert int(samples[0].value) == 1
+
+    syslog_ng.stop()
+
+
+def test_filterx_update_metric_increment(config, port_allocator, syslog_ng):
+    network_source, file_destination = create_config(
+        config,
+        port_allocator,
+        r"""
+            update_metric("const", increment=3);
+            # TODO: fix int cast from message_value
```

Comment:
Note for the future, or wanted to fix this in this PR?

Reply:
Will be fixed in https://github.com/axoflow/axosyslog/pull/229